components/input/Input.js

```diff
@@ -59,30 +59,30 @@ export default class Input extends PureComponent {
   };
 
   handleChange = event => {
-    this.updateValue(event.target.value);
+    this.updateValue(event);
   };
 
-  handleIncreaseValue = () => {
-    this.updateStep(1);
+  handleIncreaseValue = event => {
+    this.updateStep(event, 1);
   };
 
-  handleDecreaseValue = () => {
-    this.updateStep(-1);
+  handleDecreaseValue = event => {
+    this.updateStep(event, -1);
   };
 
-  updateValue(rawValue, triggerOnChange = true) {
+  updateValue(event, rawValue, triggerOnChange = true) {
     const { input, onChange } = this.props;
-    const value = Input.parseValue(rawValue, this.props);
+    const value = Input.parseValue(rawValue || event.target.value, this.props);
 
     this.setState({
       value,
     });
 
     if (triggerOnChange) {
       if (input && input.onChange) {
-        input.onChange(value);
+        input.onChange(event, value);
       } else if (onChange) {
-        onChange(value);
+        onChange(event, value);
```

Comment:
Q: Not really seeing why both parameters need to be added passed here, can you explain this please?

Reply:
If you have an input field for numbers, you can in- and decrement the number by a given step by ao clicking on a button. But as that increment value is not saved on the button element itself, it will not appear in the event object and event.target.value will be null or ' ' for that event.

Reply:
Fair enough, thanks!

 :ok: by @InstaK on 19/06/2018 12:48:21